packages/utils/src/route.js

```diff
@@ -280,3 +279,11 @@ export const promisifyRoute = function promisifyRoute (fn, ...args) {
   }
   return promise
 }
+
+export function safeEncodeComponent (str) {
+  return /%[0-9a-fA-F]{2}/.test(str) ? str : encodeURI(str)
```

Comment:
What if url includes %.. as path or parameter ?

Like: /rate/%30?max=%90

Reply:
`/%[0-9a-fA-F]{2}/.test('/rate/%30?max=%90')` > true

In this case we avoid encoding

Reply:
So if url also includes special characters, we’ll skip encoding, is that expected?

Like: /rate/%30?max=%90&name= тест

Reply:
Last commit: `/rate/%30?max=%90&name= тест` => `/rate/%30?max=%90&name=%20%D1%82%D0%B5%D1%81%D1%82`

Reply:
Last update: `/rate/%30?max=%90&name=%20%D1%82%D0%B5%D1%81%D1%82`

Reply:
Is there any merit in using something like this as our util - not reinventing url parsing?
```js
const url = '/rate/%30?max=%90&name= тест'
const { pathname, search } = new URL('http://localhost${url}')
return `${pathname}${search}`
// "/rate/%30?max=%90&name=%20%D1%82%D0%B5%D1%81%D1%82"
```

Reply:
I would like to at least use a URL aware solution instead of splitting (and finally we should move url utils to a package). But depending on environment URL, might be tricky and inconsistent. At least here is IE 11 behavior:

![image](https://user-images.githubusercontent.com/5158436/101186422-5ae3dc00-3653-11eb-85b6-c2a13123d8d7.png)

Reply:
In fact we also get benefit of leading slash normalisation if we do:
```js
function safeEncode (str) {
  const { pathname, search } = new URL(str, 'http://localhost')
  return `${pathname}${search}`
}
```

About IE - it is one of the few exceptions - could we polyfill in this case?

Reply:
Indeed we can but it costs bundle size for modern browsers and actually it is not enough to make url utils (for joining or ensure there is no leading slash for example) -- not specific to this PR's case

Reply:
In fact IE11 works fine for me - just have to wrap in brackets.
![Screenshot from 2020-12-04 16-23-20](https://user-images.githubusercontent.com/28706372/101188062-1ce3b980-364d-11eb-9788-a1bb465e2ad0.png)

Reply:
Fair enough :) Switching to URL

Reply:
There is just one downside that it is implicit it is full url or URI :))

Reply:
Thanks for bringing it up. Working on util before merge ;)